### PR TITLE
feat(testing): tiered test execution — gate-build tiers, sw-build inner-loop, build-fixer infra awareness

### PR DESCRIPTION
**Pull request by MacAttak**
## Summary

- gate-build now runs four test tiers in order (build → test → test:integration → test:smoke) with per-tier verdicts and evidence
- sw-build compressed by 318 words (-19%) and gains inner-loop validation: runs integration tests after TDD, delegates to build-fixer on failure
- Build-fixer gains infrastructure awareness — checks DB, containers, clusters before assuming code bugs

## Acceptance Criteria

| AC | Description | Status | Evidence |
|----|-------------|--------|----------|
| AC-1 | Config schema supports tiered test commands | WARN | DESIGN.md:268, gate-build:24-26 (boundary condition implicitly covered) |
| AC-2 | gate-build executes all configured tiers in order | PASS | gate-build:39-68, test-gate-build-tiers.sh (36 tests) |
| AC-3 | gate-build visibility rule for unconfigured tiers | PASS | gate-build:43-45, test-gate-build-tiers.sh:345-365 |
| AC-4 | sw-build compression (≥300 words) | PASS | 1657→1339 (-318), test-sw-build-inner-loop.sh:86-262 |
| AC-5 | sw-build inner-loop validation | PASS | sw-build:127-132, test-sw-build-inner-loop.sh:266-400 |
| AC-6 | gate-tests boundary check compression | PASS | gate-tests:52,67 |
| AC-7 | Build-fixer infrastructure awareness | PASS | build-fixer:39-43 (27 words) |
| AC-8 | Headless policy for inner-loop failure | PASS | headless.md:46,119 |
| AC-9 | Documentation surfaces updated | PASS | DESIGN.md:268, build test 182/182 |

## Blast Radius

- `core/skills/gate-build/SKILL.md` — local (gate execution only)
- `core/skills/sw-build/SKILL.md` — adjacent (build-to-verify handoff)
- `core/agents/specwright-build-fixer.md` — local (failure handling)
- `core/skills/gate-tests/SKILL.md` — local (boundary check compression)
- `core/protocols/headless.md` — local (policy row)

Does NOT change: gate-tests behavior, gate-spec, gate-semantic, gate-security, gate-wiring, sw-verify orchestration, sw-design, sw-plan, sw-ship, state management, git operations.

## Gate Results

| Gate | Status | Findings (B/W/I) |
|------|--------|-------------------|
| gate-build | PASS | 0/0/0 |
| gate-tests | PASS | 0/0/0 |
| gate-security | PASS | 0/0/0 |
| gate-wiring | PASS | 0/0/8 |
| gate-semantic | SKIP — markdown-only changes | — |
| gate-spec | PASS | 0/1/0 |

## Test Coverage

- 182 existing build tests: PASS
- 36 new gate-build tier tests: PASS
- 54 new sw-build inner-loop tests: PASS
- **Total: 272 tests, 0 failures**

## Evidence

- Build: exit 0, all tests green
- [Aggregate report](.specwright/work/testing-inner-loop/units/tiered-test-execution/evidence/aggregate-report.md)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: core/skills/gate-build/SKILL.md [hunk 1]
```diff
@@ -61,5 +80,9 @@ if the code doesn't build or tests don't pass, nothing else matters.
 |-----------|--------|
 | Build command not configured | SKIP build check, continue to test check |
 | Test command not configured | SKIP test check |
```
[the remark this answers is not in the document]
Comment (MacAttak) [reply core/skills/gate-build/SKILL.md]: Intentional. The old build/test SKIP behavior was silent (existing behavior preserved per AC-1). The INFO visibility rule is specifically for the new tiers (integration/smoke) to make the testing gap visible. The generic rule at lines 43-45 covers all tiers; the failure mode table shows per-condition actions. Updating old rows would change existing behavior.

File: core/skills/sw-build/SKILL.md [hunk 2]
```diff
@@ -137,29 +116,21 @@ When delegating, include in the prompt (in this order):
...
 **Post-build review (MEDIUM freedom):**
 - After all tasks committed, delegate review to `specwright-reviewer`.
 - Follow `protocols/build-quality.md` for trigger, depth calibration, delegation details, and findings triage.
 
+**Inner-loop validation (MEDIUM freedom) — runs after post-build review:**
+If `commands.test:integration` is configured, run it (5-minute timeout). On pass, note
+in status card. On fail, delegate to `specwright-build-fixer` (max 2 attempts) — the
+fixer should check infrastructure health before assuming code is wrong. If still failing
+after 2 attempts: interactive — present to user; headless — skip and record in
+headless-result.json. If not configured, skip silently.
```
[the remark this answers is not in the document]
Comment (MacAttak) [reply core/skills/sw-build/SKILL.md]: Intentional. Added documenting note in 9012d6d: 'verify re-runs integration tests via gate-build; the inner-loop catch is earlier, when fixer context is fresh.' Same reason real engineers run tests locally AND CI runs them again — the inner-loop catch happens while the build-fixer is in context and can fix infra issues.
